modin/core/dataframe/base/dataframe/dataframe.py

```diff
@@ -0,0 +1,541 @@
+        reduce_func : callable(row|col) -> single value, optional
+            The reduce function to apply to the results of the map function.
+        dtypes : str, optional
+            The data types for the result. This is an optimization
+            because there are functions that always result in a particular data
+            type, and this allows us to avoid (re)computing it.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the same columns as the previous, with only a single row.
+
+        Notes
+        -----
+        The user-defined function must reduce to a single value.
+
+        If the user-defined function requires access to the entire column, please use reduce instead.
+        """
+        pass
+
+    @abstractmethod
+    def infer_types(self, columns_list: List[str]) -> "ModinDataframe":
+        """
+        Determine the compatible type shared by all values in the specified columns, and coerce them to that type.
+
+        Parameters
+        ----------
+        columns_list : list of strings
+            List of column labels to infer and induce types over.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe with the inferred schema.
+        """
+        pass
+
+    @abstractmethod
+    def join(
+        self,
+        axis: Union[int, Axis],
+        condition: Callable,
+        other: "ModinDataframe",
+        join_type: str,
+    ) -> "ModinDataframe":
+        """
+        Join this dataframe with the other.
+
+        Parameters
+        ----------
+        axis : int or modin.core.dataframe.base.dataframe.Axis
+            The axis to perform the join on.
+        condition : callable
+            Function that determines which rows should be joined. The condition can be a
+            simple equality, e.g. "left.col1 == right.col1" or can be arbitrarily complex.
+        other : ModinDataframe
+            The other data to join with, i.e. the right dataframe.
+        join_type : string {"inner", "left", "right", "full"}
+            The type of join to perform.
+
+        Returns
+        -------
+        ModinDataframe
+            A new ModinDataframe that is the result of applying the specified join over the two
+            dataframes.
+
+        Notes
+        -----
+        During the join, this dataframe is considered the left, while the other is
+        treated as the right.
+
+        Only inner joins, left outer, right outer, and full outer joins are currently supported.
+        Support for other join types (e.g. natural join) may be implemented in the future.
+        """
+        pass
+
+    @abstractmethod
+    def concat(
+        self,
+        axis: Union[int, Axis],
+        others: Union["ModinDataframe", List["ModinDataframe"]],
+    ) -> "ModinDataframe":
+        """
+        Append the rows of identical column labels from multiple dataframes.
```

Comment:
It is unclear to me. What about appending the cols of identical row labels?

Reply:
Fixed!